runtime-modules/content-directory/src/lib.rs

```diff
@@ -1158,14 +1217,30 @@ decl_module! {
             property_value_vector
                 .ensure_index_in_property_vector_is_valid(index_in_property_vector)?;
 
+            let involved_entity_id = property_value_vector
+                .get_vec_value_ref()
+                .get_involved_entities()
+                .and_then(|involved_entities| involved_entities.get(index_in_property_vector as usize).copied());
+
+            // Remove value at in_class_schema_property_id in property value vector
+            // Get VecInputPropertyValue wrapped in InputPropertyValue
+            let property_value_vector_updated = Self::remove_at_index_in_property_vector(
+                property_value_vector, index_in_property_vector
+            );
+
+            let class_id = entity.get_class_id();
+
+            // Ensure provided `Property` with `unique` flag set is `unique` on `Class` level
+            Self::ensure_property_unique_option_satisfied(class_id, in_class_schema_property_id, &property_value_vector_updated.simplify())?;
+
             //
             // == MUTATION SAFE ==
             //
 
-            let involved_entity_id = property_value_vector
-                .get_vec_value()
-                .get_involved_entities()
-                .and_then(|involved_entities| involved_entities.get(index_in_property_vector as usize).copied());
+            // Insert updated propery value into entity_property_values mapping at in_class_schema_property_id.
```

Comment:
Same cleanup missing here.

Reply:
Only default non required unique value hashes can be removed, otherwise we only update its respective hash.